packages/devtools-evm-hardhat/src/tasks/healthcheck/validate-rpcs.ts

```diff
@@ -0,0 +1,179 @@
+            if (message) {
+                logger.error(message)
+            }
+            ws.close()
+            resolve(success)
+        }
+
+        ws.onopen = () => {
+            const rpcRequest = JSON.stringify({
+                jsonrpc: JSON_RPC,
+                method: 'eth_blockNumber',
+                params: [],
+                id: 1,
+            })
+            ws.send(rpcRequest)
+        }
+
+        ws.onmessage = (event) => {
+            try {
+                const response = JSON.parse(event.data.toString())
+                if (response.jsonrpc === JSON_RPC && !!response.result) {
+                    cleanup(true)
+                } else {
+                    cleanup(false, `Invalid RPC URL ${rpcUrl} for network ${networkName} response: ${event.data}`)
+                }
+            } catch (error) {
+                cleanup(false, `Error parsing RPC response for ${rpcUrl} for network ${networkName}: ${error}`)
+            }
+        }
+
+        ws.onerror = (error) => {
+            cleanup(
+                false,
+                `Validation failed for ${rpcUrl} for network ${networkName}: ${error instanceof Error ? error.message : 'An unknown error occurred'}`
+            )
+        }
+
+        ws.onclose = () => {
+            cleanup(
+                false,
+                `Validation failed for ${rpcUrl} for network ${networkName}: WebSocket connection closed unexpectedly`
+            )
+        }
+    })
+}
+
+const validateRpcUrl = async (rpcUrl: string | undefined, timeout: number, networkName: string): Promise<boolean> => {
+    if (!rpcUrl || rpcUrl.trim() === '') {
+        logger.error(`Missing RPC URL for network: ${networkName}`)
+        return false
+    }
+
+    if (rpcUrl.startsWith(HTTP_URL) || rpcUrl.startsWith(HTTPS_URL)) {
+        return await validateHttpsRpcUrl(rpcUrl, timeout, networkName)
+    } else if (rpcUrl.startsWith(WS_URL) || rpcUrl.startsWith(WSS_URL)) {
+        return await validateWebSocketRpcUrl(rpcUrl, timeout, networkName)
+    }
+
+    logger.error(`Unsupported RPC protocol in network: ${networkName}`)
+
+    return false
+}
+
+const action: ActionType<TaskArguments> = async (taskArgs, hre) => {
+    printLogo()
+
+    const networks = hre.userConfig.networks || {}
+    const eidByNetworkName = getEidsByNetworkName(hre)
+    const networkNames = Object.keys(networks).filter((networkName) => !!eidByNetworkName[networkName])
+
+    logger.info(
+        `========== Validating RPC URLs with ${taskArgs.timeout}ms timeout for networks: ${networkNames.join(', ')}`
+    )
+
+    const networksWithInvalidRPCs: string[] = []
+
+    const validationPromises = networkNames.map(async (networkName) => {
+        const rpcUrl = networks[networkName]?.[RPC_URL_KEY]
+
+        if (rpcUrl && !(await validateRpcUrl(rpcUrl, taskArgs.timeout, networkName))) {
+            networksWithInvalidRPCs.push(networkName)
+        }
+    })
+
+    await Promise.all(validationPromises)
```

Comment:
```typescript
await Promise.all(Object.entries(eidByNetworkName).map(([networkName, eid]) => {
  if (eid == null) return
  
  const provider = await providerfactory(eid);
  
  // Now do your call, maybe with some timeout like so
  return Promise.race([
    provider.getBlock(),
    // We need to finally add this to some devtools package but for now
    new Promise<void>((, reject) => setTimeout(reject, 5000))
  ]).then(
    // Block call succeeded
    block => {}
    // Error was thrown, either a timeout or the block call failed
    () => {
      // Do your fail logic
      networksWithInvalidRPCs.push(networkName)
    }
}))
```

Reply:
Gotcha -- I will format the validation promises like this in the next commit, thanks!